### ISSUE TITLE
[typeahead] reopen result dropdown on onFocus (feature request/enhancement)

### ISSUE THREAD

**Issue by joh-klein** [angular-ui/bootstrap#6046]
We have the following use case: 
* user starts typing, dropdown opens and shows results
* user clicks out of the input field (nothing is deleted from the input field), dropdown closes
* user clicks back into the typeahead input field (and does NOT start to type anything) and nothing happens. → the desired behavior is: dropdown opens again and shows the same result list as the last time (of course, that only happens, if there is anything in the input field)

Behavior is similar to the search field in firefox

**Comment by SagarNaliyapara** [comment]
i have same issue

**Comment by matthiasdailey-ccri** [comment]
I agree — this is what makes the most sense. The dropdown was visible when the user was typing before losing focus. So when regaining focus, the user would expect to see it become visible again. 

If it's undesirable because of the possible use case of the user tabbing through form fields, and the dropdowns making "too much happen", then at least we could do this: make the dropdown able to be triggered with a "Down" keypress.

**Comment by shyamal890** [comment]
+1

**Comment by lmxin123** [comment]
+1

**Comment by nicholaspoz** [comment]
+1

**Comment by dharanid** [comment]
+1
